src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -198,6 +224,83 @@ public void cancelScheduledOperations(Long programmingExerciseId) {
         instanceMessageSendService.sendProgrammingExerciseScheduleCancel(programmingExerciseId);
     }
 
+    /**
+     * validates the settings of a new programming exercise
+     *
+     * @param programmingExercise The programming exercise that should be validated
+     * @param course              The course the programming exercise should be created in or imported to
+     */
+    public void validateNewProgrammingExerciseSettings(ProgrammingExercise programmingExercise, Course course) {
+        if (programmingExercise.getId() != null) {
+            throw new BadRequestAlertException("A new programmingExercise cannot already have an ID", "Exercise", "idexists");
+        }
+
+        programmingExercise.validateGeneralSettings();
+        programmingExercise.validateProgrammingSettings();
+        programmingExercise.validateManualFeedbackSettings();
+        auxiliaryRepositoryService.validateAndAddAuxiliaryRepositoriesOfProgrammingExercise(programmingExercise, programmingExercise.getAuxiliaryRepositories());
+        submissionPolicyService.validateSubmissionPolicyCreation(programmingExercise);
+
+        ProgrammingLanguageFeature programmingLanguageFeature = programmingLanguageFeatureService.get()
+                .getProgrammingLanguageFeatures(programmingExercise.getProgrammingLanguage());
+
+        // Check if package name is set
+        if (programmingLanguageFeature.isPackageNameRequired()) {
+            if (programmingExercise.getPackageName() == null) {
+                throw new BadRequestAlertException("The package name is invalid", "Exercise", "packagenameInvalid");
+            }
+
+            // Check if package name matches regex
+            Matcher packageNameMatcher;
+            if (programmingExercise.getProgrammingLanguage() == ProgrammingLanguage.SWIFT) {
+                packageNameMatcher = packageNamePatternForSwift.matcher(programmingExercise.getPackageName());
+            }
+            else {
+                packageNameMatcher = packageNamePattern.matcher(programmingExercise.getPackageName());
+            }
```

Comment:
How about using a switch case and throwing an exception in case of an unexpected language?

Reply:
I don't quite get what you mean.
I can't think of a case where an unexpected language occurs because then the outer if statement (isPackageNameRequired) already evaluates to false.

Reply:
Currently this is the case, but the implementation of isPackageNameRequired might change but this code (accidentally) stays the same.
So explicitly handling the expected languages might be more secure